--- ./Scripts/UIScripts.lua.orig	2019-01-13 21:48:52.000000000 -0800
+++ ./Scripts/UIScripts.lua	2019-01-13 21:48:40.000000000 -0800
@@ -3650,6 +3650,22 @@
 
 		SetMenuOptions({ Name = "LocalMPScreen", Properties = { Run = "MarkTooltipDirty" } })
 
+		RiteClub = {
+			TeamA = {
+				Name = LocalMPTeamAName,
+				StartingPyreHealth = LocalMPTeamAPyreHealth,
+			},
+			TeamB = {
+				Name = LocalMPTeamBName,
+				StartingPyreHealth = LocalMPTeamBPyreHealth,
+			},
+			Rite = {
+				Stage = matchSite,
+				TalismansEnabled = GetConfigOptionValue({Name="LocalMPUseItems"}),
+				MasteriesAllowed = (characterLevel-1),
+			},
+		}
+		PersistVariable({Name = "RiteClub"})
 	end
 }
 
@@ -3715,6 +3731,7 @@
 		if GetConfigOptionValue({ Name = "LocalMPMap" }) == "LocalMP_MapRandom" then
 			SetupMapRotation()
 			mapToLoad = RemoveRandomValue( mapRotation )
+			RiteClub.Rite.Stage = mapToLoad
 			if IsEmpty( mapRotation ) then
 				mapRotation = nil
 			end
@@ -3748,6 +3765,7 @@
 	if mapRequest == "LocalMP_MapRandom" then
 		SetupMapRotation()
 		mapRequest = RemoveRandomValue( mapRotation )
+		RiteClub.Rite.Stage = mapRequest
 		if IsEmpty( mapRotation ) then
 			mapRotation = nil
 		end
