--- ./Scripts/PostMatchScripts.lua.orig	2019-01-02 10:58:03.000000000 -0800
+++ ./Scripts/PostMatchScripts.lua	2018-12-30 18:09:49.000000000 -0800
@@ -1,6 +1,7 @@
 --[[ **POST-MATCH** ]]
 
 Import "PlayerDialogueScripts.lua"
+Import "RiteClubScripts.lua"
 
 Using "StarPulse"
 Using "StarLineConnector"
@@ -363,6 +364,8 @@
 -- Post-Match Victory/Loss Presentation
 function MatchConclusionPresentation( team )
 
+  RiteClubReport(team)
+
 	HideMatchUI()
 	if raining then
 		RainStop()
